### PR TITLE
MCLOUD-5650: Magento\Framework\App\DeploymentConfig\Reader::load function is taking too much time

### PR DESCRIPTION
**Pull request by dhorytskyi**
### Description
Magento\Framework\App\DeploymentConfig\Reader::load periodically takes a lot of time.
This method is just trying to include and merge configuration from app/etc/env.php and app/etc/config.ph, so it should not take 300 seconds to be executed.
In cloud environment these two files are being shared between the servers, so concurrent & repetitive reading on files on file systems can cause delays. 

### Fixed Issues (if relevant)
https://jira.corp.magento.com/browse/MCLOUD-5650

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

releasenotes
Added fix to improve performance of the `Magento\Framework\App\DeploymentConfig\Reader::load` function, which periodically experienced long load time that affected site performance.

### CODE REVIEW

**Comment by BaDos** [comment]
Please, remove tests from the patch, also check that the path can be applied to Magento CE.

**Review by oshmyheliuk — DISMISSED** [review]

**Review by shiftedreality — DISMISSED** [review]

**Comment by shiftedreality** [comment]
@dhorytskyi, please  resolve conflict

File: patches.json [hunk 1]
```diff
@@ -229,6 +229,9 @@
     },
     "FPC is getting disabled during deployments": {
       ">=2.3.2 <2.4.0": "MAGECLOUD-5069__fpc_is_getting_disabled_during_deployments__2.3.2.patch"
```
Comment (NadiyaS) [inline patches.json]: patch for FPC was already updated in develop branch. Please merge those changes

**Review by NadiyaS — DISMISSED** [review]

File: patches.json (hunk 1 above)
Comment (dhorytskyi) [reply patches.json]: Resolved

**Review by shiftedreality — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Review by NadiyaS — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
@dhorytskyi any ideas how it can be tested?

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**